ssl/quic/quic_impl.c

```diff
@@ -3295,13 +3299,99 @@ static int qc_get_stream_avail(QCTX *ctx, uint32_t class_,
     return ret;
 }
 
+QUIC_NEEDS_LOCK
+static int qctx_should_autotick(QCTX *ctx)
+{
+    int event_handling_mode;
+
+    if (ctx->is_stream) {
+        event_handling_mode = ctx->xso->event_handling_mode;
+        if (event_handling_mode != SSL_VALUE_EVENT_HANDLING_MODE_INHERIT)
+            return event_handling_mode == SSL_VALUE_EVENT_HANDLING_MODE_IMPLICIT;
+    }
+
+    event_handling_mode = ctx->qc->event_handling_mode;
+    if (event_handling_mode == SSL_VALUE_EVENT_HANDLING_MODE_INHERIT)
+        event_handling_mode = SSL_VALUE_EVENT_HANDLING_MODE_IMPLICIT;
+
+    return event_handling_mode == SSL_VALUE_EVENT_HANDLING_MODE_IMPLICIT;
```

Comment:
Or possibly this could be `return event_handling_mode != SSL_VALUE_EVENT_HANDLING_MODE_EXPLICIT` and drop the "special case" inherit handling.